oauth2-useragent-tester/src/test/java/com/microsoft/alm/oauth2/useragent/AppTest.java

```diff
@@ -111,6 +116,40 @@ private void test_main_wiremock(final String providerName) throws Exception {
 
     }
 
+    private void test_main_native_app_wiremock(final String providerName) throws Exception {
+        final URI authorizationEndpoint = new URI(PROTOCOL, null, localHostName, wireMockPort, "/oauth2/authorize", "response_type=code&client_id=main_wiremock&state=chicken", null);
+        final URI authorizationConfirmation = new URI(PROTOCOL, null, localHostName, wireMockPort, "/oauth2/confirm", "state=chicken", null);
+        final String redirectingBody = String.format("<html><head><meta http-equiv='refresh' content='1; url=%1$s'></head><body>Redirecting to %1$s...</body></html>", authorizationConfirmation.toString());
+        final String nativeAppRedirect = "urn:ietf:wg:oauth:2.0:oob";
+        final URI redirectUri = new URI(nativeAppRedirect + "?code=steak&state=chicken");
+        stubFor(get(urlEqualTo(authorizationEndpoint.getPath() + "?" + authorizationEndpoint.getQuery()))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html")
+                        .withBody(redirectingBody)));
+        stubFor(get(urlEqualTo(authorizationConfirmation.getPath() + "?" + authorizationConfirmation.getQuery()))
+                .willReturn(aResponse()
+                        .withStatus(302)
+                        .withHeader("Location", redirectUri.toString())
+                        .withBody(redirectingBody)));
+        stubFor(get(urlEqualTo(redirectUri.getPath() + "?" + redirectUri.getQuery()))

```

Comment:
It looks like this last call to `stubFor()` is not necessary, as `redirectUri.getPath() + "?" + redirectUri.getQuery()` yields `null?null` and it should be handled by the protocol handler instead of wiremock, anyway.